connection-cache/src/connection_cache.rs

```diff
@@ -179,12 +274,27 @@ where
             eviction_timing_ms,
         } = match map.get(addr) {
             Some(pool) => {
-                if pool.need_new_connection(self.connection_pool_size) {
+                let (need_connection, create_connection_immediately) =
+                    pool.need_new_connection(self.connection_pool_size);
+                if create_connection_immediately {
                     // create more connection and put it in the pool
                     drop(map);
                     self.create_connection(&mut lock_timing_ms, addr)
                 } else {
                     let connection = pool.borrow_connection();
```

Comment:
Shouldn't the connection pool already be "warmed up" the first time it's created? In any case, I think the logic of "warming up" the connection pool should be encapsulated within the connection pool itself - an Arc to the sender can just be passed into the connection pool constructor.

Reply:
I will defer that suggested change to future PRs to limit the size of this PR.